modules/dnn/src/layers/pooling_layer.cpp

```diff
@@ -95,7 +96,8 @@ class PoolingLayerImpl CV_FINAL : public PoolingLayer
             else
                 CV_Error(Error::StsBadArg, "Unknown pooling type \"" + pool + "\"");
 
-            getPoolingKernelParams(params, kernel_size, globalPooling, pads_begin, pads_end, strides, padMode);
+            getPoolingKernelParams(params, kernel_size, isGlobalPooling, pads_begin, pads_end, strides, padMode);
+            globalPooling = std::accumulate(isGlobalPooling.begin(), isGlobalPooling.end(), 0) == 3;
```

Comment:
let's compare with `isGlobalPooling.size()` instead of hardcoded value

Reply:
Shall we check that at least one of the flags is true instead? As we do above: `is_global = globalPooling[0] || globalPooling[1] || globalPooling[2];`